scripts/tokenizer.py

```diff
@@ -4,39 +4,92 @@
 from pygments.lexers import Python3Lexer
 from pygments.token import Token
 from random import randint
+import re
 
 
 class Tokenizer:
     TOKENS = list()
+    VARS = list()
+    QUOTES = ["'", '"', '"""', "'''"]
+    BOOLS = ['True', 'False']
 
     def __init__(self, data):
+        def token_filter(tokens):
+            new_tokens = list()
+            for key, token in enumerate(tokens):
+                if tokens[key][0] == Token.Text and tokens[key][1] == ' ' and \
+                (tokens[key+1][1] == '=' or tokens[key-1][1] == '='): 
+                    pass
+                else:
+                    new_tokens.append((tokens[key][0], \
+                        re.sub('\t+', ' ', re.sub(' +', ' ', tokens[key][1]))))  # (?P<variable>.+)\s*=\s*(?P<value>.+)" # Use .strip here
+            return new_tokens
         self._tokens = Python3Lexer().get_tokens(data)
-        self.tokenize()
-
-    def genenate_id(self):
-        while True:
-            tok_id = randint(0, 1000)
-            if tok_id in [i[0] for i in self.TOKENS]:
-                continue
-            return tok_id
+        self._tokens = token_filter(token_filter(list(self._tokens))) # Avoid space char bug
+        self._tokenize()
 
     def find_by_id(self, _id):
         for token in self.TOKENS:
             if token[0] == _id:
                 return token
         return (None, None, None)
 
+    def find_index_by_id(self, _id):
+        index_list = list()
+        for index, token in enumerate(self.TOKENS):
+            if self.TOKENS[index][0] == _id:
+                index_list.append(index)
+        return index_list
+
     def get_next_token(self):
         return next(self._tokens)
 
-    def tokenize(self):
+    def get_tokens(self):
+        return self.TOKENS
+
+    def get_variables(self): # str | int | float | bool
+       return self.VARS
+
+    def _generate_id(self): # Generate ID
+        while True:
+            tok_id = randint(0, 1000)
```

Comment:
I wonder why `1000` is max?

Reply:
I assume that it is for testing purposes. (#22) Ofc, we can increase the value. And we should actually...